### ISSUE TITLE
Fix hisat2 unaligned file name extensions

### ISSUE THREAD

**Issue by mikej888** [riboviz/RiboViz#27]
I encountered an issue when trying to parse some `.sam` files in `tmp/` using `pysam`. Parsing the `tmp/WT3AT_unaligned.sam`` threw an exception:

```python
import pysam
file_name = "/home/ubuntu/RiboViz-Outputs/CentOS7/vignette-bash/tmp/WT3AT_unaligned.sam"
align_file = pysam.AlignmentFile(file_name)
```
```
ValueError: file has no sequences defined (mode='r') - is it SAM/BAM format? Consider opening with check_sq=False
```
```python
align_file = pysam.AlignmentFile(file_name, check_sq=False)
print(align_file.header)
```
```
@SRR1042864.416 HWI-ST279:248:BC0N3DACXX:1:1101:11336:2238 length=51
```
```python
print(list(align_file.header.keys()))
```
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pysam/libcalignmentfile.pyx", line 552, in pysam.libcalignmentfile.AlignmentHeader.keys
  File "pysam/libcalignmentfile.pyx", line 427, in pysam.libcalignmentfile.AlignmentHeader.to_dict
AssertionError: header line with invalid type 'SRR1042864.416 HWI-ST279:248:BC0N3DACXX:1:1101:11336:2238 length=51': '@SRR1042864.416 HWI-ST279:248:BC0N3DACXX:1:1101:11336:2238 length=51'
```

Similarly:

```bash
samtools view ~/RiboViz-Outputs/CentOS7/vignette-bash/tmp/WT3AT_unaligned.sam 
```
```
[W::sam_read1] Parse error at line 2
[main_samview] truncated file.
```

Looking at `WT3AT_unaligned.sam`:

```bash
head /home/ubuntu/RiboViz-Outputs/CentOS7/vignette-bash/tmp/WT3AT_unaligned.sam"
```
```
@SRR1042864.416 HWI-ST279:248:BC0N3DACXX:1:1101:11336:2238 length=51
TCTAGATTGCCACCAAGAGCTACCAACT
+
CCCFFFFFHHHHHJJJJJJJJJJJJJJJ
@SRR1042864.343 HWI-ST279:248:BC0N3DACXX:1:1101:9699:2191 length=51
AAAGAGGGCAATATTAGCATCAATAGTG
+
C@?DDFFFGFHHDGHJJJJIJIIIIHIG
@SRR1042864.170 HWI-ST279:248:BC0N3DACXX:1:1101:5307:2247 length=51
AAGGACGATTCTGCCA
```

This is a `.fastq` file not a `.sam` file. Checking the other `.sam` files in `tmp/` gives:

```bash
head -n 4 ~/RiboViz-Outputs/CentOS7/vignette-bash/tmp/WT*sam
```
```
==> /home/ubuntu/RiboViz-Outputs/CentOS7/vignette-bash/tmp/WT3AT_orf_map_clean.sam <==
@HD	VN:1.0	SO:unsorted
@SQ	SN:YAL068C	LN:863
@SQ	SN:YAL067W-A	LN:728
@SQ	SN:YAL067C	LN:2282

==> /home/ubuntu/RiboViz-Outputs/CentOS7/vignette-bash/tmp/WT3AT_orf_map.sam <==
@HD	VN:1.0	SO:unsorted
@SQ	SN:YAL068C	LN:863
@SQ	SN:YAL067W-A	LN:728
@SQ	SN:YAL067C	LN:2282

==> /home/ubuntu/RiboViz-Outputs/CentOS7/vignette-bash/tmp/WT3AT_rRNA_map.sam <==
@HD	VN:1.0	SO:unsorted
@SQ	SN:RDN18-1	LN:1800
@SQ	SN:RDN18-2	LN:1800
@SQ	SN:RDN25-1	LN:3396

==> /home/ubuntu/RiboViz-Outputs/CentOS7/vignette-bash/tmp/WT3AT_unaligned.sam <==
@SRR1042864.416 HWI-ST279:248:BC0N3DACXX:1:1101:11336:2238 length=51
TCTAGATTGCCACCAAGAGCTACCAACT
+
CCCFFFFFHHHHHJJJJJJJJJJJJJJJ

==> /home/ubuntu/RiboViz-Outputs/CentOS7/vignette-bash/tmp/WTnone_orf_map_clean.sam <==
@HD	VN:1.0	SO:unsorted
@SQ	SN:YAL068C	LN:863
@SQ	SN:YAL067W-A	LN:728
@SQ	SN:YAL067C	LN:2282

==> /home/ubuntu/RiboViz-Outputs/CentOS7/vignette-bash/tmp/WTnone_orf_map.sam <==
@HD	VN:1.0	SO:unsorted
@SQ	SN:YAL068C	LN:863
@SQ	SN:YAL067W-A	LN:728
@SQ	SN:YAL067C	LN:2282

==> /home/ubuntu/RiboViz-Outputs/CentOS7/vignette-bash/tmp/WTnone_rRNA_map.sam <==
@HD	VN:1.0	SO:unsorted
@SQ	SN:RDN18-1	LN:1800
@SQ	SN:RDN18-2	LN:1800
@SQ	SN:RDN25-1	LN:3396

==> /home/ubuntu/RiboViz-Outputs/CentOS7/vignette-bash/tmp/WTnone_unaligned.sam <==
@SRR1042855.154 2:1101:21136:2215:Y length=51
AGAAGAATAATGGGCGT
+
?@@DDDDDH4D8<BGIG
```

i.e. only `WTnone_unaligned.sam` and `WT3AT_unaligned.sam` don't seem to be `.sam` files. These are created in `hisat2` calls in the vignette but are not then used later (which may be why this wasn't detected):

```
hisat2 -p 4 -k 2 --no-spliced-alignment --rna-strandness F --no-unal --un vignette/tmp/WT3AT_unaligned.sam -x vignette/index/YAL_CDS_w_250 -S vignette/tmp/WT3AT_orf_map.sam -U vignette/tmp/WT3AT_nonrRNA.fq

hisat2 -p 4 -k 2 --no-spliced-alignment --rna-strandness F --no-unal --un vignette/tmp/WTnone_unaligned.sam -x vignette/index/YAL_CDS_w_250 -S vignette/tmp/WTnone_orf_map.sam -U vignette/tmp/WTnone_nonrRNA.fq
```

The vignette needs to be updated to fix the file extensions of these files.

**Comment by mikej888** [comment]
Closed by 91eb58a

**Comment by ewallace** [comment]
Good call - clearly my programming error. Aligned reads from hisat2 are output as .sam, unaligned reads are output as .fastq asnothing has been done to them.
